### ISSUE TITLE
Keyboard navigation / demo

### ISSUE THREAD

**Issue by tmcw** [mapbox/mapbox-gl-js#725]
Not sure if this should be in core or only exposed as a demo, but @bnolan's keyboard navigation demo is compelling: http://bennolan.com/experiments/walk/#17.50/-41.294436/174.772663/0
